src/py/flwr/server/superlink/fleet/grpc_rere/server_interceptor.py

```diff
@@ -133,6 +123,33 @@ def _generic_method_handler(
                         ),
                     )
                 )
+                try:
+                    node_id_from_client_public_key = self.state.get_node_id(
+                        client_public_key_bytes
+                    )
+                except KeyError:
+                    node_id_from_client_public_key = None
+
+                if node_id_from_client_public_key is not None:
+                    self.state.restore_node(
+                        node_id_from_client_public_key, request.ping_interval
+                    )
+                    self.state.store_node_id_client_public_key_pair(
+                        client_public_key_bytes, node_id_from_client_public_key
+                    )
+                    return CreateNodeResponse(
+                        node=Node(
+                            node_id=node_id_from_client_public_key, anonymous=False
+                        )
+                    )
+                response: CreateNodeResponse = method_handler.unary_unary(
+                    request, context
+                )
+                self.state.store_node_id_client_public_key_pair(
+                    client_public_key_bytes, response.node.node_id
+                )
```

Comment:
I'm wondering of we should do the initial storing of `node_id` and corresponding public key in the final method handler. Wdyt?

Reply:
Here the method handler is rerouted, so there is no default method handler call, instead of create node, restore node is being called.